xarray/tests/test_backends.py

```diff
@@ -563,6 +563,35 @@ def test_encoding_same_dtype(self):
             self.assertEqual(actual.x.encoding['dtype'], 'f4')
         self.assertEqual(ds.x.encoding, {})
 
+    @contextlib.contextmanager
+    def roundtrip_append(self, data, save_kwargs={}, open_kwargs={},
+                         allow_cleanup_failure=False):
+        with create_tmp_file(
+                allow_cleanup_failure=allow_cleanup_failure) as tmp_file:
+            for i, key in enumerate(data.variables):
+                mode = 'a' if i > 0 else 'w'
+                data[[key]].to_netcdf(tmp_file, mode=mode, **save_kwargs)
```

Comment:
This is currently always using the default backend. You need to explicitly set `engine` here somewhere, though much of the logic can potentially live on the base class or helper function of some sort. There's no magic that sets this automatically for subclasses.

Take a look at how `roundtrip()` looks. I think I led you astray by suggesting that you move everything to the baseclass.